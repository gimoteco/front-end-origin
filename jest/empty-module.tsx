import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
