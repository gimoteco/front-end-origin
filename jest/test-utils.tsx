import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
