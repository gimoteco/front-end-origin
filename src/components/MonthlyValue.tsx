import * as React from 'react';
import { formatMoney } from '../utils/format';
import * as Style from './MonthlyValue.styles';

interface MonthlyValueProps {
  value: number;
}

function MonthlyValue({ value }: MonthlyValueProps) {
  return (
    <Style.MonthlyValueWrapper>
      <Style.Label>Monthly</Style.Label>
      <Style.Value>{formatMoney(value)}</Style.Value>
    </Style.MonthlyValueWrapper>
  );
}

export default React.memo(MonthlyValue);
