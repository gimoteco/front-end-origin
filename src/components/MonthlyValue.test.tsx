import React from 'react';
import { render } from '../../jest/test-utils';
import MonthlyValue from './MonthlyValue';

describe('MonthlyValue', () => {
  it('should render according snapshot', () => {
    const { container } = render(<MonthlyValue value={200} />);

    expect(container).toMatchSnapshot();
  });
});
