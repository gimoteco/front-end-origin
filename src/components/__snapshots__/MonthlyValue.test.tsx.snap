// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyValue should render according snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa UTvpt"
  >
    <span
      class="sc-bwzfXH doWSly"
    >
      Monthly
    </span>
    <span
      class="sc-htpNat cEmdZK"
    >
      $200
    </span>
  </div>
</div>
`;
